asv/plugins/conda.py

```diff
@@ -85,30 +86,44 @@ def _matches(cls, python):
 
     def _setup(self):
         try:
-            conda = util.which('conda')
+            conda = self.find_executable('conda')
         except IOError as e:
             raise util.UserError(str(e))
 
         log.info("Creating conda environment for {0}".format(self.name))
-        util.check_call([
-            conda,
-            'create',
-            '--yes',
-            '-p',
-            self._path,
-            '--use-index-cache',
-            'python={0}'.format(self._python),
-            'pip'])
-
-        log.info("Installing requirements for {0}".format(self.name))
-        self._install_requirements(conda)
-
-    def _install_requirements(self, conda):
-        self.install('wheel')
+        # create a temporary environment.yml file
+        # and use that to generate the env for benchmarking
+
+        with tempfile.NamedTemporaryFile(mode='w', delete=False) as env_file:
+            env_file.write('name: {0}\n'
+                           'channels:\n'.format(self.name))
+            env_file.writelines(('   - %s\n' % ch for ch in self._conda_channels))
+            env_file.write('dependencies:\n'
+                           '   - python={0}\n'
+                           '   - wheel\n'
+                           '   - pip\n'.format(self._python))
+            # categorize & write dependencies based on pip vs. conda
+            reqs = self._get_requirements(conda)
+            # handle the case where there are multiple dependencies
```

Comment:
Empty line before comments block when at the same intendation as the preceding line.